src/UGrid/Mesh2D.cpp

```diff
@@ -214,9 +213,13 @@ void Mesh2D::put(ugridapi::Mesh2D const& mesh2d)
     }
 
     // Edges
-    if (auto const it = m_topology_attribute_variables.find("edge_node_connectivity"); mesh2d.edge_node != nullptr && it != m_topology_attribute_variables.end())
+    if (auto const it = m_topology_attribute_variables.find("edge_node_connectivity"); mesh2d.edge_nodes != nullptr && it != m_topology_attribute_variables.end())
```

Comment:
A general comments about put: 

1. Are all topology attributes mandatory? Are some optional?  If the key of a mandatory attribute is not found in the map, a run time error should be thrown. 
2. Can't the if statements be replaced by a generic function?